### ISSUE TITLE
[Audit log policy profiles] Could you add a profile or other mechanism to disable audit logging

### ISSUE THREAD

**Issue by dcdh** [openshift/cluster-kube-apiserver-operator#1026]
Hello,

Audit produces a lot of logs.

```
[root@control-plane-0 kube-apiserver]# ls -ltrh
total 664M
-rw-r--r--. 1 root root 100M  3 janv. 16:56 audit-2021-01-03T16-56-54.361.log
-rw-r--r--. 1 root root 100M  3 janv. 17:16 audit-2021-01-03T17-16-50.803.log
-rw-r--r--. 1 root root 100M  3 janv. 17:28 audit-2021-01-03T17-28-49.458.log
-rw-r--r--. 1 root root 100M  3 janv. 17:44 audit-2021-01-03T17-44-29.228.log
-rw-r--r--. 1 root root 100M  3 janv. 18:04 audit-2021-01-03T18-04-00.401.log
-rw-r--r--. 1 root root 100M  3 janv. 18:23 audit-2021-01-03T18-23-17.021.log
-rw-r--r--. 1 root root  43M  5 janv. 21:27 audit.log
```
More than 600M only for 1h30... it's quite huge.

Could it be possible to add a profile or other mechanism to disable the auditing feature ?

I am running OpenShift version 4.5.0-0.okd-2020-10-15-235428

Keep looking for the profile feature available in 4.6.

Damien

**Comment by sttts** [comment]
We will implement https://issues.redhat.com/browse/RFE-1448. That should solve the problem.

**Comment by dcdh** [comment]
@sttts many thanks to have created an evolution to add a less verbose custom profile by implementing an user requests profile.

I've got a question regarding how it works.

I've notice that audit log policy are defined here https://github.com/openshift/library-go/blob/master/pkg/operator/apiserver/audit/manifests/audit-policies-cm.yaml as config map.

Is it possible to manually define a custom k8s policy as a config map and reference it in the `spec.audit.profile` likes defined here https://docs.openshift.com/container-platform/4.6/security/audit-log-policy-config.html ?

**Comment by Sifa171** [comment]
@sttts the RFE doesnt sound like a on/off switch for audit logging. 
Would it be possible to implement a profile, which turns off audit logging completely? 
Thanks in advance!

**Comment by dcdh** [comment]
Like @Sifa171 said, I do not think that the RFE will respond to this request.

I am using the default profile which only log metada (metalevel ?) "Logs only metadata for read and write requests; does not log request bodies. This is the default policy."

With your new `UserRequests` audit policy it will keep metadata and add user request level on user requests. Thus, size will increase a lot.

Could you define a mechanism to allow a devops to define by himself Audit Policy (by example, defining custom ConfigMap using specific annotation or label ...).

**Comment by dcdh** [comment]
> @sttts many thanks to have created an evolution to add a less verbose custom profile by implementing an user requests profile.
> 
> I've got a question regarding how it works.
> 
> I've notice that k8s audit log policies are defined here https://github.com/openshift/library-go/blob/master/pkg/operator/apiserver/audit/manifests/audit-policies-cm.yaml as config map.
> 
> Is it possible to manually define a custom k8s policy as a config map and reference it in the `spec.audit.profile` likes defined here https://docs.openshift.com/container-platform/4.6/security/audit-log-policy-config.html ?

I reply to myself regarding adding custom policy.
`oc` refuse to validate another profile name because an enum is used for defining type of audit profile.
https://github.com/openshift/api/blob/bb81baaf35cde6491ce29f0379815cad3e3d0948/config/v1/types_apiserver.go#L62

**Comment by dcdh** [comment]
Audit is defined here https://github.com/openshift/enhancements/blob/master/enhancements/kube-apiserver/audit-policy.md

**Comment by dcdh** [comment]
@sttts After digging I think that two audits mechanisms are running in parallel.

One from `openshift-kube-api-server` and another one from `openshift-apiserver`.

Your audit log policy `Default`, `WriteRequestBodies` and `AllRequestBodies` apply for `openshift-apiserver` and produce logs into `/var/log/openshift-apiserver/`. The Default profile is fine because it does not produce a lot of log. (4MB after 2h30 running).

However, regarding `openshift-kube-api-server` a lot of logs are produces into `/var/log/kube-apiserver/` (650MB after 2h30 running).
The pod `kube-apiserver-control-plane-0` in `openshift-kube-apiserver` is setup to produces audit log.

So, why two audit mechanism running together ? Is it recommended to disable audit produces by `openshift-kube-api-server` and how to do it ? (I did not find a way to do it). I tried to manage from the `kube-apiserver-operator-xxxxxx` pod from `openshift-kube-apiserver-operator` namespace without finding how to do it.
